### PR TITLE
Resolve some memory scaling in MeshTags creation

### PR DESCRIPTION
**Pull request by garth-wells**
Remove some memory and computation that had high complexity.

### CODE REVIEW

File: cpp/dolfinx/mesh/MeshTags.h [hunk 1]
```diff
@@ -155,204 +155,231 @@ class MeshTags
...
     }
   }
 
+  // -------------------
+  // 5. Build MeshTags object
+
   return mesh::MeshTags<T>(mesh, e_dim, std::move(indices_new),
                            std::move(values_new));
-}
+} // namespace mesh
```
Comment (francesco-ballarin) [inline cpp/dolfinx/mesh/MeshTags.h:383]: This brace is closing create_meshtags, not namespace mesh :)

File: cpp/dolfinx/mesh/MeshTags.h [hunk 2]
```diff
@@ -155,202 +155,229 @@ class MeshTags
...
-/// @param[in] topology Array describing topology of tagged mesh
-///   entities. This array must be using input_global_indices from the
-///   mesh.
-/// @param[in] values Array of values to attach to mesh entities
+/// @param[in] comm The MPI communicator
+/// @param[in] mesh The Mesh that the tags are associated with
+/// @param[in] tag_cell_type Cell type of entities which are being
+///   tagged
+/// @param[in] entities 'Node' indices (using the of entities
+///   input_global_indices from the Mesh) fir vertices of for each
```
Comment (michalhabera) [inline cpp/dolfinx/mesh/MeshTags.h:163]: Typo "for"

File: cpp/dolfinx/mesh/MeshTags.h [hunk 3]
```diff
@@ -155,202 +155,229 @@ class MeshTags
...
+  if ((std::size_t)entities.rows() != values.size())
     throw std::runtime_error("Number of entities and values must match");
 
-  // Build array of topology node indices
-  std::vector<std::int64_t> topo_unique(
-      topology.data(), topology.data() + topology.rows() * topology.cols());
-  std::sort(topo_unique.begin(), topo_unique.end());
-  topo_unique.erase(std::unique(topo_unique.begin(), topo_unique.end()),
-                    topo_unique.end());
+  // Build sorted array of entity node indices
```
Comment (michalhabera) [inline cpp/dolfinx/mesh/MeshTags.h:185]: I do not see where this `nodes` is used now, is it?

Comment (garth-wells) [reply cpp/dolfinx/mesh/MeshTags.h]: Indeed, not longer used. Removed.

**Pull request merged**